server/src/test/java/org/opensearch/indices/ShardLimitValidatorTests.java

```diff
@@ -104,7 +106,54 @@ public void testUnderShardLimit() {
         assertFalse(errorMessage.isPresent());
     }
 
-    public void testValidateShardLimit() {
+    /**
+     * This test validates that system index creation succeeds
+     * even though it exceeds the cluster max shard limit
+     */
+    public void testSystemIndexCreationSucceeds() {
+        final ShardLimitValidator shardLimitValidator = createTestShardLimitService(1);
+        final Settings settings = Settings.builder()
+            .put(SETTING_VERSION_CREATED, Version.CURRENT)
+            .put(SETTING_NUMBER_OF_SHARDS, 1)
+            .put(SETTING_NUMBER_OF_REPLICAS, 1)
+            .build();
+        final ClusterState state = createClusterForShardLimitTest(1, 1, 0);
+        shardLimitValidator.validateShardLimit(".tasks", settings, state);
```

Comment:
May be validate no `ValidationException` is thrown?
Something like: `@Test(expected = Test.None.class` to annotate on top of the method.

Reply:
It seems unnecessary. Adding the annotation `@Test(expected = Test.None.class` gives redundancy warning in Intellij. So, I guess we can skip this?

Reply:
Got it. Thanks for letting me know :).